core/new-gui/src/app/common/service/user/auth.service.ts

```diff
@@ -115,8 +120,8 @@ export class AuthService {
     this.registerAutoLogout();
     this.registerAutoRefreshToken();
     return {
-      uid: this.jwtHelperService.decodeToken(token).userId,
-      name: sub,
+      uid: uid,
+      name: this.jwtHelperService.decodeToken(token).sub,
```

Comment:
you are decoding the same token twice in line 103 and 124. lets decode the token once. use a variable to refer the decoded token.